go/pkg/client/tree.go

```diff
@@ -440,9 +440,21 @@ func (c *Client) ComputeMerkleTree(ctx context.Context, execRoot, workingDir, re
 			}
 			continue
 		}
+		if i.InputDigest != "" && i.Contents != nil {
+			return digest.Empty, nil, nil, errors.New("digest and file content cannot be provided for the same virtual input")
+		}
+		entry := uploadinfo.EntryFromBlob(i.Contents)
+		if i.InputDigest != "" {
+			dg, err := digest.NewFromString(i.InputDigest)
+			if err != nil {
+				return digest.Empty, nil, nil, err
+			}
+			entry = uploadinfo.EntryFromFile(dg, normPath)
```

Comment:
Why mark it as a file blob rather than a bytes blob?

Reply:
I'm not too familiar with specifics within the sdk. Is there a behavioural difference when marking an input as file vs a blob? Which would you suggest to use in this case?

Reply:
The path field should point to the absolute local path. Virtual inputs do not exist locally by definition so they should all be of type blob.

Reply:
Hmm, but now that we are changing that definition to include two more cases:
- files that exist locally but the digest for which is already computed
- files that do not exist locally but we know their digest and it should already exist remotely

I think the implementation should take all three cases into account.

Reply:
Going to label the cases you provided to make it easier:
1. files that exist locally but the digest for which is already computed
2. files that do not exist locally but we know their digest and it should already exist remotely

(1) In this case VirtualInputs take precedence, correct? Meaning, since we ignore the local file this would technically just be treated as (2)?

Reply:
I agree. It's simpler to define just two disjoint cases:
1. bytes (no digest) that can be digested and uploaded if necessary
2. a digest (no bytes) that should exist remotely